### ISSUE TITLE
GRPO训练的效果不收敛，reward和loss等震荡严重

### ISSUE THREAD

**Issue by tangjinou** [jingyaogong/minimind#755]
按照教程的代码使用
reward model 如下
REWARD_MODEL_PATH="/mnt/sda1/modelscope_cache/models/Shanghai_AI_Laboratory/internlm2-1_8b-reward"
训练数据 如下：
data_path="/mnt/sda1/modelscope_cache/datasets/gongjy/minimind_dataset/rlaif.jsonl

其他参数如下：
# 启动4卡训练
torchrun \
    --nproc_per_node=4 \
    --master_port=29500 \
    train_grpo.py \
    --save_dir="../out" \
    --save_weight="grpo_340_context_768_hidden_size_8_layers" \
    --epochs=3 \
    --batch_size=2 \
    --learning_rate=1e-8 \
    --dtype="bfloat16" \
    --num_workers=4 \
    --accumulation_steps=3 \
    --grad_clip=0.3 \
    --log_interval=1 \
    --save_interval=10 \
    --hidden_size=768 \
    --num_hidden_layers=8 \
    --use_moe=0 \
    --thinking_ratio=0.9 \
    --data_path="/mnt/sda1/modelscope_cache/datasets/gongjy/minimind_dataset/rlaif.jsonl" \
    --from_weight="full_sft_340_context_768_hidden_size_8_layers" \
    --wandb_project="grpo_340_context_768_hidden_size_8_layers"  \
    --num_generations=12 \
    --beta=0.07 \
    --reward_model_path="$REWARD_MODEL_PATH" \
    --from_resume=0 \
    --use_compile=1 \
    --use_wandb 

然后reward值 和 loss 都震荡 如下

<img width="1242" height="611" alt="Image" src="https://github.com/user-attachments/assets/1e531d71-ac2d-4ba0-bdbf-8b16bd53f386" />

**Comment by jingyaogong** [comment]
<img width="2294" height="1246" alt="Image" src="https://github.com/user-attachments/assets/40af5e6d-f645-476e-a363-80bb9f00fd06" />

“GRPO训练的效果不收敛” 复测过了没问题，你Step太少，没有趋势

所谓震荡所谓不收敛，强调这不是有监督的Pretrain，这是RL

**Comment by tangjinou** [comment]
<img width="1573" height="622" alt="Image" src="https://github.com/user-attachments/assets/a2b72032-0811-4b2b-a5d6-735d95379491" />

1 将近1K步了，如图reward值爬坡不明显
2 训练速度非常慢，1K步耗时在5个小时+

**Comment by jingyaogong** [comment]
1. 1e-8 lr太小

**Comment by tangjinou** [comment]
嗯 改回3e-7 lr 再测试

**Comment by jingyaogong** [comment]
2. 还是得强调一下这是RL不是pretrain，“非常慢” 根据你给的group-num设置的12，1k步Rollout 12000条回答，根据你图里每条平均长度600，那么一共采了7200万token，5小时是300min，换算光采样速度是400 token/s，训练耗时都按照直接抛开不计来算了，这可是在Torch后端而非sglang，已经相当相当相当相当相当相当相当相当相当相当相当相当相当快了阿😲

**Comment by tangjinou** [comment]
<img width="1438" height="575" alt="Image" src="https://github.com/user-attachments/assets/f67eb3c0-d643-477d-b3d8-06e72fa804af" />

reward 值一直没有向上爬升 可能的原因是？ 你的效果是MOE的模型跑的么？

**Comment by jingyaogong** [comment]
nomoe，超参全部预设

**Comment by tangjinou** [comment]
嗯 按照全部预设GRPO重新跑中，预计明早能能看下
BWT
发现internlm2-1_8b-reward模型对5以上的transformer非常不友好，依赖的版本是transformers 4.41.2，而 transformers 4.41.2 对py 3.13以及以上非常不友好

**Comment by jingyaogong** [comment]
项目 `transformers==4.57.6` 没有任何问题

**Comment by tangjinou** [comment]
• transformers==4.57.6：声明支持 Python ≥3.9，但发布时（2025-11）未正式支持 3.13，最高官方测试到 3.12￼￼￼。

• Python 3.13：属于较新版本，4.57.6 能装但有兼容风险；transformers≥5.0 才正式加入 3.13 支持￼￼￼。

**Comment by tangjinou** [comment]
嗯 按照全部预设GRPO重新跑中，预计明早能能看下
BWT
发现internlm2-1_8b-reward模型对5以上的transformer非常不友好，依赖的版本是transformers 4.41.2，而 transformers 4.41.2 对py 3.13以及以上非常不友好

**Comment by tangjinou** [comment]
<img width="1195" height="603" alt="Image" src="https://github.com/user-attachments/assets/ff13d890-31ad-41bd-bf71-27a698edbc48" />


默认参数后 模型还是学到了 reward 都还是在涨
